pkgs/development/python-modules/listparser/default.nix

```diff
@@ -0,0 +1,48 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  lxml,
+  nix-update-script,
+  poetry-core,
+  pytestCheckHook,
+  requests,
+}:
+
+buildPythonPackage rec {
+  pname = "listparser";
+  version = "0.20";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "kurtmckee";
+    repo = "listparser";
+    rev = "v${version}";
+    hash = "sha256-eg9TrjDgvHsYt/0JQ7MK/uGc3KK3uGr3jRxzG0FlySg=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  propagatedBuildInputs = [
```

Comment:
```suggestion
  dependencies = [
```

Reply:
```console
$ rg 'propagatedBuildInputs =' pkgs/development/python-modules -l  | wc -l
4374
$ rg 'dependencies =' pkgs/development/python-modules -l  | wc -l
1467
```

is `dependencies` a new thing? otherwise, AFAICT `propagatedBuildInputs` would seem to be the preferred method.

Reply:
Yes, all of my suggestions are only changes towards the relatively new way of doing these. Shouldn't have any functional impact

Reply:
switched to `stdenv.mkDerivation`, where only the standard attributes are available.

Reply:
What was the reason behind the switch?

Reply:
i'm asking the opposite question ([in this thread](https://github.com/NixOS/nixpkgs/pull/271597)). packaging is only scalable because the idioms you learn from your first package apply to your second package, and soon enough it only takes 10 minutes to write a package even if your first package took a day.

that only works because we share idioms across much of nixpkgs. language-specific derivation wrappers already work against that and newer languages to nix ([zig](https://github.com/NixOS/nixpkgs/pull/241741#issuecomment-1627467997), [hare](https://github.com/NixOS/nixpkgs/pull/295155)) are trending toward stock `stdenv.mkDerivation` with a minimal amount of language-specific logic delegated to the derivation author in light of that (as little as asking them to add `nativeBuildInputs = [ zig.hook ]` to their package).

the changes i'm learning of here are contrary to all of that. it's not worth it for the 1000s of nixpkgs-maintainers-who-might-someday-want-to-package-a-python-library to learn _even more_ language-specific things whose specificity don't (yet?) seem to have much benefit to the packager.

Reply:
> switched to `stdenv.mkDerivation`, where only the standard attributes are available.

That is wrong. python packages under python-modules have to use buildPythonPackage unless they are build outputs of other packages or have special exceptions which is not the case here. That you don't like the naming doesn't free you of that.